src/test/scala/rise/autotuning.scala

```diff
@@ -119,6 +119,37 @@ class autotuning extends test_util.Tests {
   }
   // scalastyle:on
 
+
+  // function to check cycle in parameter dependencies
+  def check_no_cycle(distribution: Map[NatIdentifier,
```

Comment:
is this function only useful for testing, or should it be moved to `src/main`?

Reply:
I moved it to the config file generation of the auto tuning, where the parameter distribution is implemented.